include/unifex/scheduler_concepts.hpp

```diff
@@ -180,22 +190,22 @@ struct _schedule::sender {
 
   static constexpr bool sends_done = true;
 
-  template(
-    typename Receiver,
-    typename Scheduler =
-      std::decay_t<callable_result_t<decltype(get_scheduler), const Receiver&>>,
-    typename ScheduleSender = callable_result_t<decltype(schedule), Scheduler&>)
+  template(typename Receiver)
     (requires receiver<Receiver>)
   friend auto tag_invoke(tag_t<connect>, sender, Receiver &&r)
-      -> connect_result_t<ScheduleSender, Receiver> {
+      -> connect_result_t<
+            schedule_result_t<std::decay_t<
+                get_scheduler_result_t<const remove_cvref_t<Receiver>&>>&>,
+            Receiver> {
     auto scheduler = get_scheduler(std::as_const(r));
     return connect(schedule(scheduler), (Receiver &&) r);
```

Comment:
We could add a `std::move(scheduler)` here and avoid the need for `std::decay_t` in the return-value type calculation.